num_enum_derive/src/lib.rs

```diff
@@ -657,41 +715,45 @@ pub fn derive_try_from_primitive(input: TokenStream) -> TokenStream {
         }
 
         impl ::core::convert::TryFrom<#repr> for #name {
-            type Error = ::#krate::TryFromPrimitiveError<Self>;
+            type Error = #krate::TryFromPrimitiveError<Self>;
 
             #[inline]
             fn try_from (
                 number: #repr,
-            ) -> ::core::result::Result<Self, ::#krate::TryFromPrimitiveError<Self>>
+            ) -> ::core::result::Result<Self, #krate::TryFromPrimitiveError<Self>>
             {
-                ::#krate::TryFromPrimitive::try_from_primitive(number)
+                #krate::TryFromPrimitive::try_from_primitive(number)
             }
         }
     })
 }
 
 #[cfg(feature = "proc-macro-crate")]
-fn get_crate_name() -> String {
-    let found_crate = proc_macro_crate::crate_name("num_enum").unwrap_or_else(|err| {
-        eprintln!("Warning: {}\n    => defaulting to `num_enum`", err,);
-        proc_macro_crate::FoundCrate::Itself
-    });
-
-    match found_crate {
-        proc_macro_crate::FoundCrate::Itself => String::from("num_enum"),
-        proc_macro_crate::FoundCrate::Name(name) => name,
-    }
+fn get_crate_name(path: Option<syn::Path>) -> syn::Path {
+    path.unwrap_or_else(|| {
+        let found_crate = proc_macro_crate::crate_name("num_enum").unwrap_or_else(|err| {
+            eprintln!("Warning: {}\n    => defaulting to `num_enum`", err,);
+            proc_macro_crate::FoundCrate::Itself
+        });
+
+        match found_crate {
+            proc_macro_crate::FoundCrate::Itself => parse_quote!(::num_enum),
+            proc_macro_crate::FoundCrate::Name(name) => {
+                let krate = format_ident!("{}", name);
+                parse_quote!( :: #krate )
+            }
+        }
+    })
 }
-
 // Don't depend on proc-macro-crate in no_std environments because it causes an awkward dependency
 // on serde with std.
 //
 // no_std dependees on num_enum cannot rename the num_enum crate when they depend on it. Sorry.
 //
 // See https://github.com/illicitonion/num_enum/issues/18
 #[cfg(not(feature = "proc-macro-crate"))]
-fn get_crate_name() -> String {
-    String::from("num_enum")
+fn get_crate_name(path: Option<syn::Path>) -> syn::Path {
```

Comment:
```suggestion
fn get_crate_path(path: Option<syn::Path>) -> syn::Path {
```

Let's rename this (and the other `#[cfg`-gated one, and its call sites) to make more clear that it has the `::` prefix (and that callers don't need to add it)